File: Daily Resources/Day13/DSA - Two Pointers Solutions/python/Container With Most Water.py
from typing import List
def maxArea(height: List[int]) -> int:
    left, right, res = 0, len(height) - 1, 0
    while left < right:
        res = max(res, min(height[left], height[right]) * (right - left))
        if height[left] < height[right]:
            left += 1
        else:
            right -= 1
    return res

File: Daily Resources/Day13/DSA - Two Pointers Solutions/cpp/Container With Most Water.cpp
#include <bits/stdc++.h>
using namespace std;
int maxArea(vector<int>& height) {
    int left = 0, right = height.size() - 1, res = 0;
    while (left < right) {
        res = max(res, min(height[left], height[right]) * (right - left));
        if (height[left] < height[right]) left++;
        else right--;
    }
    return res;
}
